src/rngs/thread.rs

```diff
@@ -96,29 +96,35 @@ impl Default for ThreadRng {
 }
 
 impl ThreadRng {
+    /// Return a mutable reference to the internal RNG.
+    ///
+    /// # Safety
+    ///
+    /// The caller must make sure to stop using the returned reference before
+    /// anyone else calls this method.
```

Comment:
The function signature is `fn(&'a mut self) -> &'a mut ReseedingRng<Core, OsRng>`, which means that this is already checked by borrowck. I think @RalfJung was saying that the `unsafe` block should be documented as being same **because** of this.

Reply:
You are right, `&mut self` does not allow any further use anyway. I'll change it accordingly.

Reply:
@bjorn3
> The function signature is fn(&'a mut self) -> &'a mut ReseedingRng<Core, OsRng>, which means that this is already checked by borrowck. 

As @dhardy pointed out below, this is *not* sufficient.